src/toil/jobStores/azureJobStore.py

```diff
@@ -784,26 +825,48 @@ def attempt(last=False):
         try:
             yield
         except retriable_exceptions as e:
-            # Any instance of these exceptions is automatically retriable.
-            if last:
-                raise
+            if time.time() < expiration:
```

Comment:
It would be desirable to include the next delay in the condition. If the next delay puts us past the expiration we don't need to delay or try again.